python/metatensor-operations/metatensor/operations/_dispatch.py

```diff
@@ -610,3 +610,95 @@ def where(array):
         return np.where(array)
     else:
         raise TypeError(UNKNOWN_ARRAY_TYPE)
+
+
+def take(array, indices, axis: int):
+    """
+    See torch take_along_dim or numpy take as reference.
+
+    Because numpy and torch have different APIs we went for the more limited one
+    in torch not supporting ``mode`` argument. The argument ``out`` is not supported
+    by TorchScript, or at least it is nontrivial to add
+
+    :param array: the array the elements are returned from
+    :param indices: the indices to take long an axes
+    :param axis: axis of array to take
+
+    :return: the elments at the indices in the array along dimension specified by the
+    axis
+    """
+    if isinstance(array, TorchTensor):
+        return torch.index_select(array, dim=axis, index=indices)
+    elif isinstance(array, np.ndarray):
+        return np.take(array, indices, axis=axis)
+    else:
+        raise TypeError(UNKNOWN_ARRAY_TYPE)
+
+
+# PR COMMENT I implemented argsort because I thought I need it, but it turned out not
+#            but I would nevertheless keep it, might be useful for something else
```

Comment:
I would not keep code we don't need around, but you can post this function as a comment to the PR so we can find it easily later if we need it

Reply:
okay